### PR TITLE
chore(integrations): move sanic,snowflake,sqlalchemy,sqlite3,starlette,structlog... to internal

### PR DESCRIPTION
**Pull request by rachelyangdog**
- Moves all integration internals in ddtrace/contrib/(integration name)/ to ddtrace/contrib/internal/(integration name)/ for sanic, snowflake, sqlalchemy, sqlite3, starlette, structlog, and subprocess
- Ensures ddtrace/contrib/(integration name)/ and ddtrace/contrib/(integration name)/ continue to expose the same functions, classes, imports, and module level variables (via from ..internal.integration.module import * imports).
- Log a deprecation warning if internal modules in ddtrace/contrib/(integration name)/ and ddtrace/contrib/(integration name)/. Only patch and unpack methods should be exposed by these packages.
- https://github.com/DataDog/dd-trace-py/pull/9996

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/contrib/internal/sqlite3/patch.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+
+def get_version():
+    # type: () -> str
+    return sqlite3.sqlite_version
+
+
+def patch():
+    wrapped = wrapt.FunctionWrapper(_connect, traced_connect)
```
Comment (rachelyangdog) [inline ddtrace/contrib/internal/sqlite3/patch.py:41]:
```suggestion
    if getattr(sqlite3, "_datadog_patch", False):
          return
    sqlite3._datadog_patch = True
    wrapped = wrapt.FunctionWrapper(_connect, traced_connect)
```

File: ddtrace/contrib/internal/sqlite3/patch.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    sqlite3.connect = wrapped
+    sqlite3.dbapi2.connect = wrapped
+
+    if asm_config._iast_enabled:
+        _set_metric_iast_instrumented_sink(VULN_SQL_INJECTION)
+
+
+def unpatch():
+    sqlite3.connect = _connect
```
Comment (rachelyangdog) [inline ddtrace/contrib/internal/sqlite3/patch.py:51]:
```suggestion
    if getattr(sqlite3, "_datadog_patch", False):
        return
    sqlite3._datadog_patch = False
    sqlite3.connect = _connect
```

File: ddtrace/contrib/internal/starlette/patch.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+import starlette
+from starlette import requests as starlette_requests
+from starlette.concurrency import run_in_threadpool
+from starlette.middleware import Middleware
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace._trace.span import Span  # noqa:F401
+from ddtrace.contrib import trace_utils
+from ddtrace.contrib.asgi.middleware import TraceMiddleware
```
Comment (mabdinur) [inline ddtrace/contrib/internal/starlette/patch.py:17]:
```suggestion
from ddtrace.contrib.asgi import TraceMiddleware
```

File: ddtrace/contrib/internal/structlog/patch.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
+import structlog
+
+import ddtrace
+from ddtrace import config
+from ddtrace.contrib.logging.constants import RECORD_ATTR_ENV
+from ddtrace.contrib.logging.constants import RECORD_ATTR_SERVICE
+from ddtrace.contrib.logging.constants import RECORD_ATTR_SPAN_ID
+from ddtrace.contrib.logging.constants import RECORD_ATTR_TRACE_ID
+from ddtrace.contrib.logging.constants import RECORD_ATTR_VALUE_EMPTY
+from ddtrace.contrib.logging.constants import RECORD_ATTR_VERSION
```
Comment (mabdinur) [inline ddtrace/contrib/internal/structlog/patch.py:10]:
```suggestion
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_ENV
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_SERVICE
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_SPAN_ID
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_TRACE_ID
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_VALUE_EMPTY
from ddtrace.contrib.internal.logging.constants import RECORD_ATTR_VERSION
```

**Review by mabdinur — APPROVED** [review]

File: ddtrace/contrib/internal/subprocess/patch.py [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+from typing import List  # noqa:F401
+from typing import Optional  # noqa:F401
+from typing import Tuple  # noqa:F401
+from typing import Union  # noqa:F401
+from typing import cast  # noqa:F401
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.contrib import trace_utils
+from ddtrace.contrib.subprocess.constants import COMMANDS
```
Comment (mabdinur) [inline ddtrace/contrib/internal/subprocess/patch.py:20]:
```suggestion
from ddtrace.contrib.internal.subprocess.constants import COMMANDS
```

**Review by erikayasuda — APPROVED** [review]

**Review by wconti27 — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Comment by rachelyangdog** [comment]
@DataDog/asm-python Could I get a review please! thank you!!

**Review by gnufede — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Comment by gnufede** [comment]
You have some failing test related to fastapi: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/69176/workflows/34e792bd-3f5f-4fed-bd93-5c3b2ac25065/jobs/4225327?invite=true#step-105-11429_26

**Pull request merged**